The test for java.util.Date.toString is time zone-dependent

Which, of course, means it fails on any machine that doesn't live in CET:
```scala
[scala-js] [info] Test org.scalajs.testsuite.javalib.util.DateTest.toStringTest started
[scala-js] [error] Test org.scalajs.testsuite.javalib.util.DateTest.toStringTest failed: java.lang.AssertionError: expected:<Mon Nov 03 06:23:27 .+ 1997> to match:<Mon Nov 03 05:23:27 GMT 1997>, took 0.007000064 sec
[scala-js] [error] at <jscode>.(file:/home/jenkins/workspace/scala-2.11.x-integrate-community-build/target-0.9.5/project-builds/scala-js-b949c9e7573510c8b79f7ab6d6b38cfef57747fb/junit-runtime/src/main/scala/org/junit/Assert.scala:28)
[scala-js] [error] at <jscode>.(file:/home/jenkins/workspace/scala-2.11.x-integrate-community-build/target-0.9.5/project-builds/scala-js-b949c9e7573510c8b79f7ab6d6b38cfef57747fb/junit-runtime/src/main/scala/org/junit/Assert.scala:14)
[scala-js] [error] at <jscode>.(file:/home/jenkins/workspace/scala-2.11.x-integrate-community-build/target-0.9.5/project-builds/scala-js-b949c9e7573510c8b79f7ab6d6b38cfef57747fb/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/DateTest.scala:93)
[scala-js] [error] at <jscode>.(file:/home/jenkins/workspace/scala-2.11.x-integrate-community-build/target-0.9.5/project-builds/scala-js-b949c9e7573510c8b79f7ab6d6b38cfef57747fb/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/DateTest.scala:94)
[scala-js] [error] at <jscode>.(Lorg_scalajs_testsuite_javalib_util_DateTest$scalajs$junit$bootstrapper$.sjsir:90)
[scala-js] [error] ... 8 more
[scala-js] [info] Test run finished: 1 failed, 0 ignored, 10 total, 0.572s
```
See https://github.com/scala-js/scala-js/pull/2567#issuecomment-242864873

--- a/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/DateTest.scala
+++ b/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/DateTest.scala
@@ -91,10 +91,10 @@ class DateTest {
   @Test def toStringTest(): Unit = {
     def test(expectedRegex: String, actual: String): Unit =
       assertTrue(s"expected:<$expectedRegex> to match:<$actual>", actual.matches(expectedRegex))
-    test("Mon Nov 03 06:23:27 .+ 1997", new Date(Date.UTC(97, 10, 3, 5, 23, 27)).toString)
-    test("Sun Dec 31 01:00:00 .+ 1899", new Date(Date.UTC(0, 0, 0, 0, 0, 0)).toString)
-    test("Sun Jan 05 09:01:09 .+ 1902", new Date(Date.UTC(1, 12, 5, 8, 1, 9)).toString)
-    test("Sat Jan 09 06:03:04 .+ 2900", new Date(Date.UTC(1000, 0, 9, 5, 3, 4)).toString)
+    test("Mon Nov 03 05:23:27 .+ 1997", new Date(97, 10, 3, 5, 23, 27).toString)
+    test("Sun Dec 31 00:00:00 .+ 1899", new Date(0, 0, 0, 0, 0, 0).toString)
+    test("Sun Jan 05 08:01:09 .+ 1902", new Date(1, 12, 5, 8, 1, 9).toString)
+    test("Sat Jan 09 05:03:04 .+ 2900", new Date(1000, 0, 9, 5, 3, 4).toString)
   }
 
   @Test def toGMTString(): Unit = {

